src/coordinator/ts/m3db/block/block_iteration_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package block
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func makeResults() [][]float64 {
+	var results [][]float64
+	sliceOfNaNs := []float64{math.NaN(), math.NaN()}
+	results = append(results, sliceOfNaNs)
+	results = append(results, []float64{1, 1})
+	results = append(results, sliceOfNaNs, sliceOfNaNs, sliceOfNaNs)
+	results = append(results, []float64{3, 3})
+	for i := 0; i < 11; i++ {
+		results = append(results, sliceOfNaNs)
+	}
+	results = append(results, []float64{5, 5})
+	results = append(results, sliceOfNaNs)
+	results = append(results, []float64{6, 6})
+
+	return results
+}
+
+func TestStepIteration(t *testing.T) {
```

Comment:
probably need more tests, also should write down conditions you are testing. This test is pretty opaque